### ISSUE TITLE
404 /favicon.ico when deployed as ROOT.war

### ISSUE THREAD

**Issue by HiromuHota** [HiromuHota/pentaho-kettle#46]
How to reproduce:
```
$ rm -rf webapps/*
$ cp spoon.war webapps/ROOT.war
$ ./bin/startup.sh
```

You will get a 404 error after successful login.
![image](https://cloud.githubusercontent.com/assets/20668349/25100271/4fd3c6e6-2364-11e7-954c-82562df5941c.png)

Workaround: try again by typing the url in the address bar (e.g., `http://localhost:8080/spoon`).

**Comment by HiromuHota** [comment]
According to http://forum.spring.io/forum/spring-projects/security/93678-redirect-to-favicon-ico, this can be resolved by adding `<intercept-url pattern="favicon.ico" access="permitAll"/>` to WEB-INF/spring/security.xml

**Comment by HiromuHota** [comment]
Resolved by dcb2697bd9e8926400d0f40370b570dfdb3e3f47.
